packages/widget-utils/README.md

```diff
@@ -19,14 +19,56 @@ To embed a widget in the website, just add a `<div>` container and decorate it w
 #### Attributes
 
 - `data-tocco-widget-key` (required): The key of the widget config
+- `data-id` (optional): An id to be able to reference events and methods through `window` obj
 
-### Example
+#### Example
 
 ```html
 <div data-tocco-widget-key="12"></div>
 <script src="https://${customer.tocco.ch}/js/tocco-widget-utils/dist/bootstrap.js"></script>
 ```
 
+### Events
+A widget can provide events which will be invoked. When a widget gets rendered the event handlers which are defined on the `window.toccoEventHandlers` will be attached to the widget events.
+
+Use an flat object of functions which will be used for each widget on that page. Preferably only used if only one widget is emebedded.
+```js
+window.toccoEventHandlers = {
+  loginSuccess: () => {}
+}
+```
+
+Use the `data-id` attribute to define the property name on the `toccoEventHandlers` obj to defined event handlers for each widget. Preferable used if multiple widgets are embedded.
+```html
+<div data-id="login" data-tocco-widget-key="12"></div>
+```
+
+```js
+window.toccoEventHandlers = {
+  login: {
+    loginSuccess: () => {}
+  }
+}
+```
+
+### Methods
+A widget can provide methods which can be invoked to communicate with the widget itself.
+When a widget gets rendered these methods will be attached to `window.toccoMethods` to provide them to outer scripts.
+
+The `data-id` attribute is used as a property name on the `window.toccoMethods` object. If the `data-id` attribute is not set the methods will not be attached to the `toccoMethods` object.
+
+```html
+<div data-id="login" data-tocco-widget-key="12"></div>
+```
+```js
+window.toccoMethods = {
+  login: {
+    setLocale: () => {}
+    ...
+  }
+}
```

Comment:
This `toccoMethods` obj is the structure we're going to set up for the methods. To me it looks like the user should provide this `toccoMethods` object.

I'd replace it with (or at least add) an example of how they can be called.

Like for example:

```js
// e.g. update locale of the login widget
window.toccoMethods.login.setLocale('en')
```